drivers/kodi/device.js

```diff
@@ -28,80 +28,113 @@ class KodiDevice extends Homey.Device {
 
     onDeleted() {
         this.log('deleted()')
-        // Remove listeners
-        if (this.kodi.socket) {
-            this._kodi.socket.removeAllListeners()
+        this.cleanup();
+    }
+
+    cleanup() {
+        if (this.reconnectTimer) {
+            clearTimeout(this.reconnectTimer);
```

Comment:
clearInterval()

When the kodi device gets disconnected, it needs to **periodically** poll, not just once. As far as I know setTimeout() only fires once

Reply:
True, but it keeps rescheduling the timeout if it can't connect. So basically:
1. try to connect
2. if connection fails: wait 10 seconds using `setTimeout` and go to step 1
3. if connection works, remove any scheduled timeouts
4. if connection works but fails at some point (for instance, when Kodi is shut down): wait 10 seconds using `setTimeout` and go to step 1

Typically, I prefer `setTimeout` because if, for some reason, a connection attempt takes longer than `CONNECT_INTERVAL`, the code could start multiple concurrent connection attempts if you use `setInterval`.

The [original code](https://github.com/MarvinSchenkel/HomeyKodi/blob/SDK2/drivers/kodi/device.js#L65) also uses `setTimeout` (but, confusingly, uses [`clearInterval`](https://github.com/MarvinSchenkel/HomeyKodi/blob/SDK2/drivers/kodi/device.js#L37) to cancel it).

Reply:
Yeah you are right. Not sure why I used clearInterval, probably a mistake during the port. If you could just prefix the private functions with _ then I will merge this PR :)

Reply:
Prefixes added in bb1b7aa